lib/policy_machine.rb

```diff
@@ -90,16 +90,19 @@ def add_association(user_attribute_pe, operation_set, object_attribute_pe)
   # TODO: add option to ignore policy classes to allow consumer to speed up this method.
   # TODO: Parallelize the two component checks
   def is_privilege?(user_or_attribute, operation, object_or_attribute, options = {})
-    is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_attribute, options) &&
-      (options[:ignore_prohibitions] || !is_privilege_ignoring_prohibitions?(user_or_attribute, PM::Prohibition.on(operation), object_or_attribute, options))
+    # Fixes situation where options hash's values are deleting after first is_privilege_ignoring_prohibitions? call.
+    options_copy = options.deep_dup
+    (options[:ignore_prohibitions] || !is_privilege_ignoring_prohibitions?(user_or_attribute, PM::Prohibition.on(operation), object_or_attribute, options)) &&
+      is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_attribute, options_copy)
```

Comment:
Preference for passing the copy into the prohibitions method if that's what's being mutated.
```suggestion
    (options[:ignore_prohibitions] || !is_privilege_ignoring_prohibitions?(user_or_attribute, PM::Prohibition.on(operation), object_or_attribute, options_copy)) &&
      is_privilege_ignoring_prohibitions?(user_or_attribute, operation, object_or_attribute, options)
```